src/pal/src/eventprovider/CMakeLists.txt

```diff
@@ -0,0 +1,37 @@
+cmake_minimum_required(VERSION 2.8.12.2)
+
+if(CMAKE_SYSTEM_NAME STREQUAL Linux)
+project(eventprovider)
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)   
+  
+  add_definitions(-DPAL_STDCPP_COMPAT=1)
+  include_directories(${COREPAL_SOURCE_DIR}/inc/rt)
+  add_compile_options(-fPIC)
+
+
+   add_library(eventprovider1
+"lttngeventprovMicrosoft_Linux_DotNETRuntime.cpp"
```

Comment:
The file naming convention in PAL / CoreCLR is all lowercase characters and no underscores. Do we need to have these files different for each OS in such a way that we cannot handle it by a bunch of ifdefs? If it is really the case, then I'd suggest creating a subfolder for each supported OS (and maybe even for the lltng if we plan to support other providers later) and using short names.